store/displayPreferences.ts

```diff
@@ -134,170 +122,100 @@ export const actions: ActionTree<
         }
       );
 
-      if (response.status !== 200)
+      if (response.status !== 200) {
         throw new Error(
           'get display preferences status response = ' + response.status
         );
+      }
 
-      await dispatch('initStateSuccess', { displayPreferences: response.data });
+      const data = castResponse(response.data as ClientPreferences);
```

Comment:
Why the cast as our casted interface? What does happen to `darkMode` which is still a string? Shouldn't the `castResponse` method take a DisplayPreferences and return a new ClientPreferences?

Reply:
Because that means that I need to create a copy of the object as TypeScript will complain that I can't assign a boolean to a string type.

What about I create a copy of the object in that function and "edit" that copied object instead of the passed one in-place as it's done right now?

Reply:
I don't have any problem with the inplace edit, it's just that I don't understand how that works when the cast occurs. Like darkMode is a string but it gets casted to boolean right? What happens? Does it become always true then>